R/cchart.R

```diff
@@ -299,6 +302,34 @@ CChart <- function(chart.type, x, small.multiples = FALSE,
     result
 }
 
+addWarning <- function(x, chart.type, small.multiples, has.annotations)
+{
+    export.type <- attr(x, "ChartType")
+    msg <- ""
+
+    if (small.multiples)
+        msg <- "This visualization is a small multiple which is not supported by Microsoft."
+    else if (has.annotations)
+        msg <- "This visualization contains annotations which is not supported by Microsoft."
+    else if (chart.type %in% c("Palm", "Stream", "Venn", "Pyramid",
+            "BarPictograph", "StackedColumnWithStatisticalSignificance"))
+        msg <- "This visualization type is not supported by Microsoft."
+    else if (export.type %in% c("Sunburst", "Histogram", "Filled Map",
+            "Box & Whisker"))
+        msg <- "This visualization type cannot be exported to PowerPoint."
+    # The charts in the last condition have chart types supported by powerpoint 2016,
+    # however they are not supported by the API used by Displayr
+
+    if (nzchar(msg))
+        attr(x, "ChartWarning") <- paste(msg,
+            "It will be exported to PowerPoint as an image.",
+            "Use 'PowerPoint Export Type' to change this to a",
+            "supported chart type or set to 'Image' to",
+            "suppress this warning.")
```

Comment:
I tried to follow the suggestions from the generated warning but the warning still appears as an attribute in the output. From what I can see on master the control is in the properties tab and inside the `Powerpoint Update` Group. Inside that group there is a `Format` Control. This is the closest control I could see that matches the `'PowerPoint Export Type'`. Changing that control doesn't suppress the warning. i.e. an attribute for ChartWarning is still generated since that attribute is created solely based on the viz type and doesnt check the `'PowerPoint Export Type'` control